index.js

```diff
@@ -72,26 +73,37 @@ export default class Channel {
   /**
    * Call all listeners with specified params
    */
-  dispatch () {
+  dispatch (...args) {
     if (!this._mute) {
       this._listeners.forEach(listener => {
-        listener.callback.apply(listener.context, arguments);
+        listener.callback.apply(listener.context, args);
       });
+    } else if (this._accumulate) {
+      this._accumulate.push(args);
     }
   }
 
   /**
    * Mute channel
+   * @param {Object} [options]
+   * @param {Boolean} [options.accumulate]
    */
-  mute () {
+  mute (options = {}) {
     this._mute = true;
+    if (options.accumulate) {
+      this._accumulate = [];
```

Comment:
I suggest to use two variables instead of  dynamically changing type of `this._accumulate` from `Boolean` to `Array`. 
For example `this._accumulatedEvents = []`.